autoload/firenvim.vim

```diff
@@ -857,64 +884,86 @@ function! firenvim#install(...) abort
                         continue
                 endif
 
-                try
-                        let l:manifest_content = l:cur_browser['manifest_content'](l:execute_nvim_path)
-                        let l:manifest_dir_path = l:cur_browser['manifest_dir_path']()
-                        let l:manifest_path = s:build_path([l:manifest_dir_path, 'firenvim.json'])
-                catch /.*/
-                        echo 'Aborting installation for ' . l:name . '. ' . v:exception
-                        continue
-                endtry
-
-                if has('win32') || s:is_wsl
-                        let l:manifest_path = s:build_path([l:manifest_dir_path, 'firenvim-' . l:name . '.json'])
+                let l:flatpak_result = s:browser_has_flatpak_installed(l:cur_browser)
+                if l:flatpak_result == 1
+                    try
+                            let l:manifest_content = l:cur_browser['manifest_content'](l:execute_nvim_path)
+                            let l:manifest_dir_path = l:cur_browser['manifest_dir_path']()
+                            let l:manifest_path = s:build_path([l:manifest_dir_path, 'firenvim.json'])
+                    catch /.*/
+                            echo 'Aborting installation for ' . l:name . ' flatpak. ' . v:exception
+                            continue
+                    endtry
+
+                    call s:maybe_execute('mkdir', l:manifest_dir_path, 'p', 0700)
+                    call s:maybe_execute('writefile', [l:manifest_content], l:manifest_path)
+                    call s:maybe_execute('setfperm', l:manifest_path, 'rw-------')
+
+                    echo 'Installed flatpak manifest for ' . l:name . '.'
+
+                   call s:maybe_execute('system', ['ln', '-nsf', stdpath("config"), s:build_path([$HOME, '.var', 'app', l:cur_browser['flatpak_name'], "config", "nvim"])]) 
+                   call s:maybe_execute('system', ['ln', '-nsf', stdpath("data"), s:build_path([$HOME, '.var', 'app', l:cur_browser['flatpak_name'], "data", "nvim"])]) 
+                else
```

Comment:
It seems that most of the code in the `if l:flatpak_result` branch is shared with the code in the `else` branch. What do you think about having a single execution path and only gating the two `system` calls behind an `if l:flatpak_result`? e.g.
```vim
endtry

call s:maybe_execute('mkdir', l:manifest_dir_path, 'p', 0700)
call s:maybe_execute('writefile', [l:manifest_content], l:manifest_path)
call s:maybe_execute('setfperm', l:manifest_path, 'rw-------')

echo 'Installed manifest for ' . l:name . '.'

if s:browser_has_flatpak_installed(l:cur_browser)
    call s:maybe_execute('system', ['ln', '-nsf', stdpath("config"), s:build_path([$HOME, '.var', 'app', l:cur_browser['flatpak_name'], "config", "nvim"])]) 
    call s:maybe_execute('system', ['ln', '-nsf', stdpath("data"), s:build_path([$HOME, '.var', 'app', l:cur_browser['flatpak_name'], "data", "nvim"])]) 
endif

if has('win32') || s:is_wsl
    ...
```

Reply:
That does actually make the most sense yes